spec/std/range_spec.cr

```diff
@@ -301,6 +356,49 @@ describe "Range" do
     it "is not empty with ... and begin.succ == end" do
       (1...2).reverse_each.to_a.should eq([1])
     end
+
+    it "raises on endless range" do
```

Comment:
Can we have a spec for the reverse iterator working too?

Reply:
Unless I'm misunderstanding something, this is the spec for the reverse iterator. Or did you mean something else?

Reply:
This is the spec for the `reverse_each` iterator raising, but not actually working (calling `.next`)

Reply:
Ooooh... you are right! In fact this wasn't working. I added a spec and also fixed the code.